### ISSUE TITLE
Blazor InputSelect should support Nullable<Enum>

### ISSUE THREAD

**Issue by sven5** [dotnet/aspnetcore#13624]
The `InputSelect` should support Nullable enums. In this way the initial value could be null and the user must select one value. In this way, validation works simple with the `Required` attribute.

example:
```
<InputSelect class="custom-select" id="@nameof(ImportRun.Type)" @bind-Value="@_model.Type">
    <option>Select...</option>
    <option value="@(TypeOfImport.Create)">Create</option>
    <option value="@(TypeOfImport.Charge)">Charge</option>
    <option value="@(TypeOfImport.Difference)">Set Difference</option>
</InputSelect>
```

Model:
```
[Required]
public TypeOfImport? Type { get; set; }
```

And enum is:
   ```
 public enum TypeOfImport
    {
        [Display(Name = "Create new")]
        Create = 0,

        [Display(Name = "Set difference")]
        Difference = 1,

        [Display(Name = "Charge")]
        Charge = 2
    }
```

Currently this is not working and throwing an exception. Only not-nullable enum is currently supported. Unfortunately, this preselects always the first enum value, because this is the default.

**Comment by kevingates** [comment]
Any updates on this? Honestly, this what I am running into. We have clients, and we have an Enum for Suffix fields. This field could be null, or Sr., Jr., I, II, III, etc.

If no value is entered, we want it null in our database. it would be great if Blazor could handle this common situation.

**Comment by kevingates** [comment]
Only thing I would add is that I would want to have to add the "Required" attribute. I may want the value to be null if no data is entered.

**Comment by nheath99** [comment]
This is also an issue for Nullable<int> type. This is a show-stopper for me and is preventing me porting an existing application to Blazor. Is there a work-around for this? I'm pretty disappointed it's been put back to .net 5 as it seems like such a basic requirement.

**Comment by pranavkm** [comment]
> Is there a work-around for this?

Would having a default `None` option that is mapped to `null` work? The other alternative would be to author a custom `InputSelect` component that supports nullable. @GuillaumeNury sent us a PR that should be available in 5.0-preview2. You could use that as a basis for a custom component in 3.1.

**Comment by PatPat1567** [comment]
If anyone else looks into this for a work around. You can easily take the .cs file from https://github.com/dotnet/aspnetcore/blob/master/src/Components/Web/src/Forms/InputSelect.cs and convert it to a razor component. It takes a little bit to understand how blazor can render the syntax, but can be generate yourself.
